action.yaml

```diff
@@ -19,29 +19,41 @@ inputs:
 runs:
   using: 'composite'
   steps:
+    - name: Check is Trivy already installed?
+      id: check
+      shell: bash
+      run: |
+        if [ "$TRIVY_INSTALLED" == "${{ inputs.version }}-${{ inputs.path }}" ]; then
+          echo "Trivy '${{ inputs.version }}' has already been installed by the current job, skipping reinstalling it again"
+          echo "installed=true" >> $GITHUB_OUTPUT
+        else
+          echo "installed=false" >> $GITHUB_OUTPUT
+        fi
+
     - name: Binary dir
+      if: ${{ steps.check.outputs.installed == 'false' }}
       id: binary-dir
       shell: bash
       run: echo "dir=${{ inputs.path }}/trivy-bin" >> $GITHUB_OUTPUT
 
     ## Don't cache `latest` version
     - name: Check the version for caching
-      if: ${{ inputs.cache == 'true' && inputs.version == 'latest' }}
+      if: ${{ steps.check.outputs.installed == 'false' && inputs.cache == 'true' && inputs.version == 'latest' }}
       shell: bash
       run: |
-        echo "'setup-trivy' doesn't currently support caching the `latest` version"
+        echo "'setup-trivy' doesn't currently support caching the 'latest' version"
```

Comment:
**NB** - You had backticks in the echo'd output here which caused an implicit sub shell execution in bash and was causing spurious errors e.g. 
![Screenshot 2024-10-14 at 11 20 21](https://github.com/user-attachments/assets/deef5d9b-e6a3-4f87-94f3-b65c7b7f6080)

Reply:
fixed in #8